## Q1. What are the three measures of central tendency?

`Answer`

There are three main measures of central tendency:
- mode.
- median.
- mean.

## Q2. What is the difference between the mean, median, and mode? How are they used to measure the central tendency of a dataset?

`Answer`

- The `mean` is the average where the sum of all the numbers is divided by the total number of numbers, 
- the `median` is the middle value in the list of given numbers numerically ordered from smallest to biggest 
- `mode` is the value of the number which occurs most often in the list.

These measures help summarize the central tendency of a dataset, but they focus on different aspects. The mean is influenced by all the data points and provides an average value. The median is less affected by extreme values and represents the middle value. The mode focuses on the most frequently occurring value(s) in the dataset.

## Q3. Measure the three measures of central tendency for the given height data:
[178,177,176,177,178.2,178,175,179,180,175,178.9,176.2,177,172.5,178,176.5]

In [1]:
import numpy as np
data= [178,177,176,177,178.2,178,175,179,180,175,178.9,176.2,177,172.5,178,176.5]
np.mean(data)

177.01875

In [2]:
np.median(data)

177.0

In [7]:
from scipy import stats
stats.mode(data)

/tmp/ipykernel_95/2800027621.py:2: FutureWarning: Unlike other reduction functions (e.g. `skew`, `kurtosis`), the default behavior of `mode` typically preserves the axis it acts along. In SciPy 1.11.0, this behavior will change: the default value of `keepdims` will become False, the `axis` over which the statistic is taken will be eliminated, and the value None will no longer be accepted. Set `keepdims` to True or False to avoid this warning.
  stats.mode(data)


ModeResult(mode=array([177.]), count=array([3]))

## Q4. Find the standard deviation for the given data:
[178,177,176,177,178.2,178,175,179,180,175,178.9,176.2,177,172.5,178,176.5]

In [8]:
data2= [178,177,176,177,178.2,178,175,179,180,175,178.9,176.2,177,172.5,178,176.5]
np.std(data2)

1.7885814036548633

## Q5. How are measures of dispersion such as range, variance, and standard deviation used to describe the spread of a dataset? Provide an example.

`Answer`

1. Range: The range is the simplest measure of dispersion and represents the difference between the maximum and minimum values in a dataset. It gives a rough estimate of the spread but does not provide detailed information about the distribution. For example, if you have a dataset of exam scores ranging from 60 to 90, the range would be 90 - 60 = 30.

2. Variance: Variance is a measure of the average squared deviation from the mean. It provides a more comprehensive understanding of the spread compared to the range. To calculate the variance, you take each data point, subtract the mean, square the result, sum up all the squared differences, and divide by the total count. Variance is denoted by σ² (sigma squared).

3. Standard Deviation: The standard deviation is the square root of the variance. It is the most commonly used measure of dispersion as it is in the same unit as the original data. The standard deviation indicates the average amount of deviation or dispersion of data points from the mean. A smaller standard deviation suggests that the data points are closely packed around the mean, while a larger standard deviation indicates a more spread-out distribution. Standard deviation is denoted by σ (sigma).


Let's consider an example: Suppose you have a dataset of five test scores: 70, 75, 80, 85, and 90.

In [9]:
# First, calculate the mean:
Mean = (70 + 75 + 80 + 85 + 90) / 5 
Mean

80.0

In [13]:
# Next, calculate the variance:
Variance = ((70 - 80)**2 + (75 - 80)**2 + (80 - 80)**2 + (85 - 80)**2 + (90 - 80)**2) / 5
Variance

50.0

In [17]:
# Finally, calculate the standard deviation:
import math
Standard_Deviation = math.sqrt(50)
Standard_Deviation

7.0710678118654755

In this example, the range is 90 - 70 = 20, the variance is 50, and the standard deviation is approximately 7.07. These measures describe the spread of the dataset around the mean.

## Q6. What is a Venn diagram?

`Answer`

A Venn diagram illustrates the relationships between two or more data sets. Venn diagrams are especially useful for highlighting similarities and differences and are commonly used to compare and contrast the characteristics of different data sets. In a Venn diagram, circles are used to represent each data set.

## Q7. For the two given sets A = (2,3,4,5,6,7) & B = (0,2,6,8,10). Find:
(i) A ∩ B<br>
(ii) A ⋃ B<br>

In [19]:
A = {2,3,4,5,6,7}
B = {0,2,6,8,10}
# Intersection (A ∩ B)
intersection = A.intersection(B)
intersection

{2, 6}

In [21]:
# Union (A ⋃ B)
union = A.union(B)
union

{0, 2, 3, 4, 5, 6, 7, 8, 10}

## Q8. What do you understand about skewness in data?

`Answer`

1. Positive Skewness (Right Skewness): If the tail of the distribution is skewed to the right, it indicates positive skewness. In other words, the majority of the data is concentrated on the left side of the distribution, and the right tail is longer. The mean is typically greater than the median in positively skewed distributions.

2. Negative Skewness (Left Skewness): If the tail of the distribution is skewed to the left, it indicates negative skewness. In this case, the majority of the data is concentrated on the right side of the distribution, and the left tail is longer. The mean is typically less than the median in negatively skewed distributions.

3. Zero Skewness (Symmetric Distribution): If the distribution is symmetric, it indicates zero skewness. The data is evenly distributed on both sides of the distribution, and the mean is equal to the median.

## Q9. If a data is right skewed then what will be the position of median with respect to mean?

`Answer`

for a right-skewed distribution(positively skewed):

- The mean is typically greater than the median.
- The median tends to be closer to the lower values, while the mean is influenced by the higher values in the tail.

## Q10. Explain the difference between covariance and correlation. How are these measures used in statistical analysis?

`Answer`

1. Covariance: Covariance is an indicator of the extent to which 2 random variables are dependent on each other. A higher number denotes higher dependency.<br>
The value of covariance lies in the range of -∞ and +∞.

2. Correlation: Correlation is a statistical measure that indicates how strongly two variables are related.<br>
Correlation is limited to values between the range -1 and +1

In statistical analysis, both covariance and correlation are used to understand the relationship between variables. However, correlation is more commonly used due to its standardized nature and ease of interpretation. It allows for comparisons between different datasets and enables researchers to make more accurate predictions or draw conclusions about the relationship between variables. Covariance, on the other hand, is useful in cases where the scale of the variables is important and when studying the joint variability of two variables.

## Q11. What is the formula for calculating the sample mean? Provide an example calculation for a dataset.

`Answer`

The formula for calculating the sample mean (often denoted as "x-bar") is:

Sample Mean (x-bar) = (Sum of all values in the dataset) / (Number of values in the dataset)

To illustrate with an example, let's consider the following dataset of exam scores: 78, 85, 92, 88, 79.

To calculate the sample mean, we add up all the values and divide by the total number of values:

Sum of all values = 78 + 85 + 92 + 88 + 79 = 422

Number of values = 5

Sample Mean (x-bar) = 422 / 5 = 84.4

Therefore, the sample mean of the given dataset is 84.4.

## Q12. For a normal distribution data what is the relationship between its measure of central tendency?

`Answer`

For a normal distribution, three common measures of central tendency are used: the mean, median, and mode. In a normal distribution, these measures have the following relationship:

1. Mean: The mean of a normal distribution is located at the center of the distribution. It is equal to the median and the mode. Mathematically, the mean is calculated by summing all the data points and dividing by the total number of data points.

2. Median: The median of a normal distribution is also located at the center. It is equal to the mean and the mode. The median is the middle value when the data is arranged in ascending or descending order.

3. Mode: The mode of a normal distribution is the value that appears most frequently. In a normal distribution, the mode is equal. to the mean and the median since the distribution is symmetric.

In summary, for a normal distribution, the mean, median, and mode are all equal and represent the center of the distribution. This is due to the symmetric nature of the normal distribution, where the data is evenly distributed around the mean

## Q13. How is covariance different from correlation?

`Answer`

Covariance and correlation are two terms that are opposed and are both used in statistics and regression analysis. Covariance shows you how the two variables differ, whereas correlation shows you how the two variables are related

## Q14. How do outliers affect measures of central tendency and dispersion? Provide an example.

`Answer`

1. Central Tendency:-
When a outlier is present it can effect the shape of the graph, if we have outliers to the right of the graph. These outliers are causing the mean to increase, but if we have outliers to the left of the graph these outliers are dragging down the mean. This causes a conflict because the mean no longer provides a good representation of the data, alternatively we would much rather use the median. The median on the other hand is less likely to be affect by outliers.

    1. Mean: Significant change
        - Mean increases with high outlier
        - Mean decreases with low outlier
    2. Median: Little change
    3. Mode: No change

2. dispersion:-
Variance and Standard Deviation: Both variance and standard deviation measure the spread of the data points around the mean. Outliers can significantly impact these measures, as they introduce large deviations from the mean, resulting in inflated values for variance and standard deviation.